### PR TITLE
Update algorithm docs

### PR DESCRIPTION
**Pull request by MikeOverkamp-diag**

### CODE REVIEW

File: docs/algorithms.rst [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+Algorithm Inputs
+================
+
+Inputs to an algorithm are available in the container in the ``/input`` directory. You can add :ref:`component interfaces <components>` that will provide the inputs to your algorithm.
+
+The ``relative_path`` property on the ``ComponentInterface`` is used to determine where the input value will be placed inside the container.
+
+The simple and annotation types get written to a json file, located in ``/input/{relative_path}`` in the container. The file and image types get placed in ``/input/{relative_path}/{filename}``.
+
+When creating a new experiment for an algorithm, you can fill out provide values for all ``ComponentInterfaces`` using the form provided.
```
Comment (miriam-groeneveld) [inline docs/algorithms.rst:19]:
```suggestion
When creating a new experiment for an algorithm, you can provide values for all ``ComponentInterfaces`` using the form provided.
```

File: docs/algorithms.rst [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+Any output for both `stdout` and `stderr` is captured. The output for `stderr` gets marked as a warning in the experiment's result.
+
+If an algorithm does not properly run, it should exit with a non zero exit code. The job for the algorithm then gets marked as failed.
+
+
+Algorithm Outputs
+=================
+
+Outputs of an algorithm must be stored in the directory ``/output/``. As with the inputs, a ``ComponentInterface`` needs to be defined for each of the expected outputs.
+
```
Comment (miriam-groeneveld) [inline docs/algorithms.rst:61]:
```suggestion
For outputs of kind ``JSON file`` you can determine (only when creating the ``ComponentInterface``) if the json should be stored as file or in the database. Use this option if you expect the json file to be very large.
```

**Review by miriam-groeneveld — DISMISSED** [review]
LGTM, some additions :).

File: docs/algorithms.rst [hunk 3]
```diff
@@ -0,0 +1,65 @@
+======================================
+ Algorithms
+======================================
+
+What is an algorithm?
+=====================
+
+An algorithm takes a set of inputs and a container image. A container is created from the image to process the inputs to outputs.
```
Comment (jmsmkn) [inline docs/algorithms.rst:8]:
```suggestion
An algorithm is a container image that is executed on a set of inputs, producing a set of outputs. The inputs and outputs can be text, numbers, booleans, medical images or annotations.
```

File: docs/algorithms.rst [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+* .jpeg
+* .jpg
+
+The following file formats can be uploaded and will be converted to tif:
+
+* Aperio (.svs)
+* Hamamatsu (.vms, .vmu, .ndpi)
+* Leica (.scn)
+* MIRAX (.mrxs)
+* Ventana (.bif)
```
Comment (jmsmkn) [inline docs/algorithms.rst:45]:
```suggestion
Grand Challenge works with two image formats that will need to be read or written by your algorithm container image ``.mha`` and ``.tif``.
```

File: docs/algorithms.rst [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+An algorithm takes a set of inputs and a container image. A container is created from the image to process the inputs to outputs.
+
+Algorithm Inputs
+================
+
+Inputs to an algorithm are available in the container in the ``/input`` directory. You can add :ref:`component interfaces <components>` that will provide the inputs to your algorithm.
+
+The ``relative_path`` property on the ``ComponentInterface`` is used to determine where the input value will be placed inside the container.
+
+The simple and annotation types get written to a json file, located in ``/input/{relative_path}`` in the container. The file and image types get placed in ``/input/{relative_path}/{filename}``.
```
Comment (jmsmkn) [inline docs/algorithms.rst:17]: Files do not get renamed now, made a change for that.

```suggestion
The simple and annotation types get written to a json file, located in ``/input/{relative_path}`` in the container. The image types get placed in ``/input/{relative_path}/{filename}``.
```

File: docs/algorithms.rst [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+If an algorithm does not properly run, it should exit with a non zero exit code. The job for the algorithm then gets marked as failed.
+
+
+Algorithm Outputs
+=================
+
+Outputs of an algorithm must be stored in the directory ``/output/``. As with the inputs, a ``ComponentInterface`` needs to be defined for each of the expected outputs.
+For outputs of kind ``JSON file`` you can determine (only when creating the ``ComponentInterface``) if the json should be stored as file or in the database. Use this option if you expect the json file to be very large.
+results.json
```
Comment (jmsmkn) [inline docs/algorithms.rst:62]:
```suggestion

results.json
```

**Comment by jmsmkn** [comment]
Looks great! I made a few suggestions as this should be for algorithm developers. I think we should add the `ComponentInterfaceList` page, I will do that. 

@MikeOverkamp-diag please update your poetry install with `poetry self update` as the lockfile format has changed.

**Review by jmsmkn — APPROVED** [review]

**Pull request merged**